File: engine/engine-model/src/main/scala/tools/packet/buddy/RequestAddBuddy.scala
package tools.packet.buddy

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class RequestAddBuddy(private var _characterIdFrom: Int, private var _characterIdTo: Int,
                      private var _characterNameFrom: String) extends PacketInput {
  def characterIdFrom: Int = _characterIdFrom

  def characterIdTo: Int = _characterIdTo

  def characterNameFrom: String = _characterNameFrom

  override def opcode(): SendOpcode = SendOpcode.BUDDY_LIST
}

File: engine/engine-model/src/main/java/tools/packet/buddy/RequestAddBuddy.java
package tools.packet.buddy;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record RequestAddBuddy(Integer characterIdFrom, Integer characterIdTo,
                              String characterNameFrom) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.BUDDY_LIST;
   }
}
